packages/firestore/src/core/target.ts

```diff
@@ -786,6 +786,15 @@ export class CompositeFilter extends Filter {
   }
 }
 
+// TODO(orquery) move compositeFilterWithAddedFilters to filter.ts in future refactor
+export function compositeFilterWithAddedFilters(
```

Comment:
This function shouldn't always perform `CompositeOperator.AND`. It should use the same operator as the given `compositeFilter` and add `otherFilters` to it.

Please add a comment for the function as well. ([Reference](https://cs.opensource.google/firebase-sdk/firebase-android-sdk/+/master:firebase-firestore/src/main/java/com/google/firebase/firestore/core/CompositeFilter.java;l=117-124?q=withadded&ss=firebase-sdk%2Ffirebase-android-sdk))

Reply:
Fixed. TY.